### PR TITLE
Improve how to generate peer fl context

### PR DESCRIPTION
**Pull request by IsaacYangSLA**
### Description

This PR adds a utility function which generates peer fl context in a more efficient way and update a few locations with this function

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [x] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/private/fed/server/server_commands.py [hunk 1]
```diff
@@ -180,8 +179,9 @@ def process(self, data: Shareable, fl_ctx: FLContext):
         shareable.set_header(key=FLContextKey.TASK_ID, value=task_id)
 
         shareable.set_header(key=ServerCommandKey.TASK_NAME, value=taskname)
-        shared_fl_ctx = FLContext()
-        shared_fl_ctx.set_public_props(copy.deepcopy(get_serializable_data(fl_ctx).get_all_public_props()))
+
+        # TODO
```
Comment (YuanTingHsieh) [inline nvflare/private/fed/server/server_commands.py:183]: @IsaacYangSLA what is the TODO here about?

**Review by YuanTingHsieh — COMMENTED** [review]
only one question otherwise LGTM

File: nvflare/private/fed/server/server_commands.py (hunk 1 above)
Comment (yanchengnv) [reply nvflare/private/fed/server/server_commands.py:183]: I think this PR just addressed this "TODO".

**Review by yanchengnv — APPROVED** [review]
Looks great!

File: nvflare/private/fed/server/server_commands.py (hunk 1 above)
Comment (IsaacYangSLA) [reply nvflare/private/fed/server/server_commands.py:183]: Yan is right.  I put that TODO when searching for places to update as reminders.

**Pull request merged**